commercetools/commercetools-sdk-java-api/src/integrationTest/java/commercetools/cart/CartAddressesTests.java

```diff
@@ -0,0 +1,82 @@
+
+package commercetools.cart;
+
+import static commercetools.cart.CartsFixtures.withUpdateableCart;
+import static commercetools.product.ProductFixtures.withPublishedProduct;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import com.commercetools.api.models.cart.Cart;
+import com.commercetools.api.models.cart.CartAddItemShippingAddressActionBuilder;
+import com.commercetools.api.models.cart.CartDraft;
+import com.commercetools.api.models.cart.CartDraftBuilder;
+import com.commercetools.api.models.common.BaseAddress;
+import com.commercetools.api.models.common.BaseAddressBuilder;
+import commercetools.utils.CommercetoolsTestUtils;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class CartAddressesTests {
+
+    @Test
+    public void createCartWithMultipleItemShippingAddresses() {
+        withPublishedProduct(product -> {
+            final BaseAddress address1 = BaseAddressBuilder.of()
+                    .country(Locale.GERMANY.getCountry())
+                    .key(CommercetoolsTestUtils.randomKey())
+                    .build();
+            final BaseAddress address2 = BaseAddressBuilder.of()
+                    .country(Locale.FRANCE.getCountry())
+                    .key(CommercetoolsTestUtils.randomKey())
+                    .build();
+            final BaseAddress address3 = BaseAddressBuilder.of()
+                    .country(Locale.ITALY.getCountry())
+                    .key(CommercetoolsTestUtils.randomKey())
+                    .build();
+
+            final List<BaseAddress> baseAddresses = Arrays.asList(address1, address2, address3);
+
+            final CartDraft cartDraft = CartDraftBuilder.of()
+                    .itemShippingAddresses(baseAddresses)
+                    .currency("EUR")
+                    .country("DE")
+                    .build();
+
+            final Cart createdCart = CommercetoolsTestUtils.getProjectApiRoot()
```

Comment:
Please add a step to delete the cart after the test.

Another option could be to extend the withCart fixture to adjust the cart draft before creating the cart itself

Reply:
In the PS team we tried to not delete the resources after tests. There are couple of benefits of not deleting the resources:
1. in the integration tests we want to test the integration with CT and we want simulate the real CT project if possible. In the real project you do not delete your carts afterwards and it's never an empty project.
2. it's faster for the tests if we don't delete the data. If the project starts to get full, we can always remove the whole project and recreate it.

Of course this requires a bit more work from our side so that we can create resources that are not conflicting with the existing ones.

WDYT?